packages/@sanity/types/src/slug/types.ts

```diff
@@ -2,42 +2,50 @@ import type {SlugSchemaType} from '../schema'
 import type {SanityDocument} from '../documents'
 import type {Path} from '../paths'
 
+/** @internal */
 export interface Slug {
   _type: 'slug'
   current: string
 }
 
+/** @public */
 export type SlugParent = Record<string, unknown> | Record<string, unknown>[]
 
+/** @public */
 export interface SlugSourceOptions {
   parentPath: Path
   parent: SlugParent
 }
 
+/** @public */
 export type SlugSourceFn = (
   document: SanityDocument,
   options: SlugSourceOptions
 ) => string | Promise<string>
 
+/** @public */
 export type SlugifierFn = (source: string, schemaType: SlugSchemaType) => string | Promise<string>
 
 // TODO: De-dupe with validation types
+/** @public */
 export interface SlugUniqueOptions {
   parent: SlugParent
   type: SlugSchemaType
   document: SanityDocument
-  defaultIsUnique: UniqueCheckerFn
+  defaultIsUnique: SlugUniqueCheckerFn
```

Comment:
FYI this will conflict with the slug-context changes. Interface is merged with validation types (as per the todo above)

Reply:
I see, will revert this 👍